### PR TITLE
feat(category): move category into nuxt structures WIP

### PR DESCRIPTION
**Pull request by mkucmus**

### CODE REVIEW

File: packages/default-theme/layouts/default.vue [hunk 1]
```diff
@@ -1,33 +1,20 @@
 <template>
   <div>
-    <div id="nav">
```
Comment (patzick) [inline packages/default-theme/layouts/default.vue]: please revert this file, for now, we will have here navigation and probably breadcrumbs

File: packages/shopware-6-client/src/index.ts [hunk 2]
```diff
@@ -1,12 +1,14 @@
 import { ClientSettings, setupConfig, updateConfig } from "./settings";
 import { reloadConfiguration } from "./apiService";
 
-export { config } from "./settings";
+export { config } from "./settings";import { Cart } from "../interfaces/models/checkout/cart/Cart";
```
Comment (patzick) [inline packages/shopware-6-client/src/index.ts]: I believe this import is unnecessary here
```suggestion
export { config } from "./settings";
```

File: packages/shopware-6-client/src/services/navigationService.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import {
+  getNavigationEndpoint
+} from "../endpoints";
+import { apiService } from "../apiService";
+
+interface NavigationResponse {
```
Comment (patzick) [inline packages/shopware-6-client/src/services/navigationService.ts]:
```suggestion
export interface NavigationResponse {
```

File: packages/shopware-6-client/src/services/navigationService.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  count: number,
+  elements: NaviationElement[]
+}
+
+interface NaviationElement {
```
Comment (patzick) [inline packages/shopware-6-client/src/services/navigationService.ts]:
```suggestion
export interface NavigationElement {
```

File: packages/shopware-6-client/src/services/navigationService.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import {
+  getNavigationEndpoint
+} from "../endpoints";
+import { apiService } from "../apiService";
+
+interface NavigationResponse {
+  count: number,
+  elements: NaviationElement[]
```
Comment (patzick) [inline packages/shopware-6-client/src/services/navigationService.ts]:
```suggestion
  elements: NavigationElement[]
```

File: packages/shopware-6-client/src/services/navigationService.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+interface NaviationElement {
+  id: string
+  path: string
+  name: string
+  children: NaviationElement[] | null,
```
Comment (patzick) [inline packages/shopware-6-client/src/services/navigationService.ts]:
```suggestion
  children: NavigationElement[] | null,
```

File: packages/shopware-6-client/src/services/navigationService.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  count: number,
+  level: number,
+  extensions: any[]
+}
+
+
+export async function getNavigation(
```
Comment (patzick) [inline packages/shopware-6-client/src/services/navigationService.ts]: always try to take params as an object, then it's easy to extend and change the interface without breaking changes. Also, order of params doesn't matter then.

File: packages/default-theme/layouts/default.vue (hunk 1 above)
Comment (mkucmus) [reply packages/default-theme/layouts/default.vue]: moved to another component and used here

**Pull request merged**